### PR TITLE
fix(core): do not update completed mutations

### PR DESCRIPTION
**Pull request by gautemo**
fixes https://github.com/TanStack/query/issues/6839

This PR fixes that updating options on a mutation will update the latest mutation in the cache, even though it is completed.

### CODE REVIEW

File: packages/query-core/src/tests/mutationObserver.test.tsx [hunk 1]
```diff
@@ -169,4 +169,128 @@ describe('mutationObserver', () => {
 
     unsubscribe()
   })
+
+  test('changing mutation meta should not affect successful mutations', async () => {
```
[the remark this answers is not in the document]
Comment (gautemo) [reply packages/query-core/src/tests/mutationObserver.test.tsx:173]: I see, added a test `mutation cache should have different meta when updated between mutations` in the latest commit. Also changed meta property names from `count` to `a`.

[the remark this answers is not in the document]
Comment (TkDodo) [reply packages/query-core/src/tests/mutationObserver.test.tsx:173]: `a` was just an example, you didn't need to rename it 😂

**Review by TkDodo — APPROVED** [review]

**Comment by TkDodo** [comment]
please fix formatting

**Comment by gautemo** [comment]
> please fix formatting

Sorry about that, should be fixed now

**Pull request merged**